clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -88,7 +112,7 @@ public NetworkClient(Selectable selector,
                          int socketReceiveBuffer,
                          int requestTimeoutMs,
                          Time time) {
-        this(null, metadata, selector, clientId, maxInFlightRequestsPerConnection,
+        this(null, null, metadata, selector, clientId, maxInFlightRequestsPerConnection,
```

Comment:
the nulls are getting a bit hard to read. Do we want constants for NO_APIS and NO_METADATA?